### PR TITLE
[Enhancement](page cache) insert into setting to disable page cache

### PR DESCRIPTION
**Pull request by zzzzzzzs**
## Proposed changes

insert into setting to disable page cache

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zzzzzzzs** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/InsertIntoTableCommand.java [hunk 1]
```diff
@@ -102,6 +102,8 @@ public void setJobId(long jobId) {
 
     @Override
     public void run(ConnectContext ctx, StmtExecutor executor) throws Exception {
+        // insert into setting to disable page cache
+        ctx.getSessionVariable().enablePageCache = false;
```
Comment (yiguolei) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/InsertIntoTableCommand.java:106]: If you modify the code like this. when the insert into select command finished. and user want to execute select command, the page cache value is false too. The performance maybe bad.
I think you could refer the code insert into xxx select /* SET_VAR=xxx*/
to know how to set the session variable at sql level not at session level.

**Review by yiguolei — DISMISSED** [review]
LGTM

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/EliminateLogicalSelectHint.java [hunk 2]
```diff
@@ -85,7 +90,15 @@ private void setVar(SelectHintSetVar selectHint, StatementContext context) {
         SessionVariable sessionVariable = context.getConnectContext().getSessionVariable();
         // set temporary session value, and then revert value in the 'finally block' of StmtExecutor#execute
         sessionVariable.setIsSingleSetVar(true);
-        for (Entry<String, Optional<String>> kv : selectHint.getParameters().entrySet()) {
+        Map<String, Optional<String>> parameters = selectHint.getParameters();
+        StatementBase parsedStatement = context.getParsedStatement();
+        if (parsedStatement instanceof LogicalPlanAdapter) {
+            LogicalPlan logicalPlan = ((LogicalPlanAdapter) parsedStatement).getLogicalPlan();
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/EliminateLogicalSelectHint.java:96]: should not impl here. please refer to `org.apache.doris.nereids.processor.pre.TurnOffPipelineForDml`

**Review by morrySnow — CHANGES_REQUESTED** [review]

**Review by yiguolei — DISMISSED** [review]
LGTM

**Comment by yiguolei** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/processor/pre/TurnOffPipelineForDml.java [hunk 3]
```diff
@@ -53,6 +58,15 @@ private void turnOffPipeline(StatementContext context) {
         // set temporary session value, and then revert value in the 'finally block' of StmtExecutor#execute
         sessionVariable.setIsSingleSetVar(true);
         try {
+            StatementBase parsedStatement = context.getParsedStatement();
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/processor/pre/TurnOffPipelineForDml.java:61]: not change this file, u should add a new file to do this thing

**Review by morrySnow — CHANGES_REQUESTED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/processor/pre/TurnOffPipelineForDml.java (hunk 3 above)
Comment (zzzzzzzs) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/processor/pre/TurnOffPipelineForDml.java]: done

**Comment by zzzzzzzs** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/processor/pre/TurnOffPageCacheForInsertIntoSelect.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+    @Override
+    public Plan visitUnboundTableSink(UnboundTableSink<? extends Plan> unboundTableSink,
+            StatementContext context) {
+        turnOffPageCache(context);
+        return unboundTableSink;
+    }
+
+    @Override
+    public Plan visitLogicalFileSink(LogicalFileSink<? extends Plan> fileSink, StatementContext context) {
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/processor/pre/TurnOffPageCacheForInsertIntoSelect.java:48]: only process file sink? so table sink is not turn off?

Comment (zzzzzzzs) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/processor/pre/TurnOffPageCacheForInsertIntoSelect.java:44]: done

**Comment by zzzzzzzs** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Comment by morrySnow** [comment]
run feut

**Review by yiguolei — APPROVED** [review]
lgtm

**Pull request merged**